### DISCUSSION TITLE
__init__() got an unexpected keyword argument 'cachedir'

### DISCUSSION THREAD

**Discussion by avafor** [MaartenGr/BERTopic#722] [Q&A]
Hi,
I have been working with my virtual environment in which I use BERTopic for a while. After I reinstalled my envoronment now I am not able to import BERtopic anymore. Below is the error: 

`
import bertopic
`
`TypeError                                 Traceback (most recent call last)
/tmp/ipykernel_27049/879401823.py in <module>
     20 from sklearn.feature_extraction.text import CountVectorizer
     21 from sklearn.feature_extraction.text import TfidfVectorizer
---> 22 from hdbscan import HDBSCAN
     23 from umap import UMAP
     24 from sklearn.cluster import KMeans

~/anaconda3/envs/bertopic/lib/python3.7/site-packages/hdbscan/__init__.py in <module>
----> 1 from .hdbscan_ import HDBSCAN, hdbscan
      2 from .robust_single_linkage_ import RobustSingleLinkage, robust_single_linkage
      3 from .validity import validity_index
      4 from .prediction import (approximate_predict,
      5                          membership_vector,

~/anaconda3/envs/bertopic/lib/python3.7/site-packages/hdbscan/hdbscan_.py in <module>
    507     leaf_size=40,
    508     algorithm="best",
--> 509     memory=Memory(cachedir=None, verbose=0),
    510     approx_min_span_tree=True,
    511     gen_min_span_tree=False,

TypeError: __init__() got an unexpected keyword argument 'cachedir'
`

The Bertopic version I am using is 0.11.0

Thank you for your support.

**Comment by MaartenGr** [comment]
Ah, based on your error message, the [recent commit](https://github.com/scikit-learn-contrib/hdbscan/pull/563) in the HDBSCAN repo, and that joblib was recently [updated](https://pypi.org/project/joblib/#history), it seems that the problem should be resolved by either installing HDBSCAN from its main branch or by installing a previous version of joblib:

 `pip install --upgrade joblib==1.1.0`

Hopefully, this should fix your issue!

**Reply by awdejong** [reply]
I had the same problem and this solved it! Thanks!

**Reply by senghongg** [reply]
I had the same problem and this solved it too. Thanks so much Maarten
